lib/phoenix/live_dashboard/live/chart_component.ex

```diff
@@ -64,16 +68,20 @@ defmodule Phoenix.LiveDashboard.ChartComponent do
     metric.name
     |> List.last()
     |> Phoenix.Naming.humanize()
-    |> Kernel.<>("#{humanize_unit(metric.unit)}")
+    |> Kernel.<>("#{label_suffix(metric.unit)}")
   end
 
-  defp humanize_unit(:byte), do: " (bytes)"
-  defp humanize_unit(:kilobyte), do: " (KB)"
-  defp humanize_unit(:megabyte), do: " (MB)"
-  defp humanize_unit(:nanosecond), do: " (ns)"
-  defp humanize_unit(:microsecond), do: " (µs)"
-  defp humanize_unit(:millisecond), do: " (ms)"
-  defp humanize_unit(:second), do: " s"
-  defp humanize_unit(:unit), do: ""
-  defp humanize_unit(unit) when is_atom(unit), do: " (#{unit})"
+  defp label_suffix(:second), do: " #{chart_unit(:second)}"
+  defp label_suffix(:unit), do: ""
+  defp label_suffix(unit) when is_atom(unit), do: " (#{chart_unit(unit)})"
+
+  defp chart_unit(:byte), do: "bytes"
+  defp chart_unit(:kilobyte), do: "KB"
+  defp chart_unit(:megabyte), do: "MB"
+  defp chart_unit(:nanosecond), do: "ns"
+  defp chart_unit(:microsecond), do: "µs"
```

Comment:
since uPlot relies on JS's `Date` objects for [calendar, timezone & dst-aware] `time: true` scales, it cannot get you down sub-millisecond resolution. there is currently no way to dynamically switch from `time: true` to `time: false` (to start treating x data as numbers rather than timestamps).

my "interesting" work-around was to use two plots - one with `time: true` and another `time: false`, then toggle between them in when zooming/unzooming or doing `setDate()` to different precision ranges. (the sub-ms plot needs different x data since it can't use unix or js timestamps for x anyways). more in https://github.com/leeoniya/uPlot/issues/111.

food for thought in case you foresee having `time: true` data ranges in a single plot which must zoom down to sub-ms resolutions.

Reply:
@leeoniya Thanks for the note!  FYI, these are the units for the y-axis values. I expect we'll always be using time scale for the x-axis.

Reply:
ah, ok then 👍